src/main/java/database/ArangoHandler.java

```diff
@@ -2,103 +2,117 @@
+        return dbConnection;
+    }
+
+    public ArangoDB getArangoDriver() {
+        return arangoDriver;
+    }
+
 
     /**
      * Connects to ArangoDB
      */
     public void connect(){
-        /* Read config file */
-        try{
-            ConfigReader config = new ConfigReader("config");
-            String databaseName = config.getConfig("db.name");
-            String collectionName = config.getConfig("collection.users.name");
-
-            /* Select driver */
-            driver = DatabaseConnection.getDBConnection().getArangoDriver();
-
-            /* If database does not exist */
-            if (!driver.getDatabases().contains(databaseName)) {
-                /* Create database */
-                driver.createDatabase(databaseName);
-                /* Create collection */
-                driver.db(databaseName).createCollection(collectionName);
-                System.out.println("Database created");
-            }
-
-            /* Select database */
-            database = driver.db(databaseName);
-            /* Select collection */
-            collection = database.collection(collectionName);
-        } catch (IOException e){
-            e.printStackTrace();
+        String databaseName = configProps.getProperty("db.name");
+        String collectionName = configProps.getProperty("collection.users.name");
+
+        /* Select driver */
+        arangoDriver = getDBConnection().getArangoDriver();
+
+        /* If database does not exist */
+        if (!arangoDriver.getDatabases().contains(databaseName)) {
+            /* Create database */
+            arangoDriver.createDatabase(databaseName);
+            /* Create collection */
+            arangoDriver.db(databaseName).createCollection(collectionName);
+            System.out.println("Database created");
         }
 
+        /* Select database */
+        database = arangoDriver.db(databaseName);
+        /* Select collection */
+        collection = database.collection(collectionName);
     }
 
     /**
      * Disconnects from ArangoDB
      */
     public void disconnect() {
         /* Tear down connection */
-        driver.shutdown();
-    }
-
-    /**
-     * Retrieves user profile by key
-     */
-    public UserProfile getUserProfile(String key) {
-        /* Retrieve user profile */
-        return collection.getDocument(key, UserProfile.class);
+        arangoDriver.shutdown();
     }
 
     /**
      * Creates user profile
      */
-    public String createUserProfile(UserProfile userProfile) throws ArangoDBException {
-        /* Check validity before insertion */
-        if (!isValidUserProfile(userProfile)) {
-            throw new RuntimeException("Invalid user profile");
-        }
+    public String createUser(Object user) {
+        UserProfile userProfile= (UserProfile) user;
 
-        /* Insert user profile */
         DocumentCreateEntity insertedUserProfile = collection.insertDocument(userProfile);
```

Comment:
What happens if the user already exists?, Do we need to handle an exception?

Reply:
Handled at signup. however we can only check existence(hopefully)

Reply:
I can't